### PR TITLE
Flat Earth Hexagonal

### PR DESCRIPTION
**Pull request by FiretronP75**
Add new map shape Flat Earth Hexagonal

### CODE REVIEW

**Comment by FiretronP75** [comment]
FlatEarchArchipelago
FlatEarthFourCorners
FlatEarthPangaea
FlatEarthThreeContinents

File: core/src/com/unciv/logic/map/mapgenerator/MapGenerator.kt [hunk 1]
```diff
@@ -596,6 +651,146 @@ class MapGenerator(val ruleset: Ruleset) {
...
+
+        // Flat Earth needs a 1 tile wide perimeter of ice/mountain/snow and a 2 radius cluster of ice in the center.
+        if (tileMap.mapParameters.shape === MapShape.flatEarth) {
+
+            // Make sure center tiles and edge tiles are ocean not coast
+            for (tile in tileMap.values) {
+                val isCenterTile = tile.latitude == 0f && tile.longitude == 0f
+                val isEdgeTile = tile.neighbors.count() < 6
+
+                if (isCenterTile || isEdgeTile) {
```
Comment (yairm210) [inline core/src/com/unciv/logic/map/mapgenerator/MapGenerator.kt:663]: Please flip some if these ifs to reduce nesting...

File: core/src/com/unciv/logic/map/mapgenerator/MapGenerator.kt [hunk 2]
```diff
@@ -596,6 +651,146 @@ class MapGenerator(val ruleset: Ruleset) {
                     -1f, ruleset.modOptions.constants.spawnIceBelowTemperature,
                     0f, 1f))
             }.toList()
+
+        // Flat Earth needs a 1 tile wide perimeter of ice/mountain/snow and a 2 radius cluster of ice in the center.
+        if (tileMap.mapParameters.shape === MapShape.flatEarth) {
```
Comment (yairm210) [inline core/src/com/unciv/logic/map/mapgenerator/MapGenerator.kt:656]: You have 150 lines of uninterrupted code here.
Not only does this need its own function, it probably needs subfunctions, maybe its own class

File: core/src/com/unciv/logic/map/mapgenerator/MapGenerator.kt (hunk 1 above)
Comment (FiretronP75) [reply core/src/com/unciv/logic/map/mapgenerator/MapGenerator.kt:663]: Like `if (!isCenterTile && !isEdgeTile) continue` ?

Comment (yairm210) [reply core/src/com/unciv/logic/map/mapgenerator/MapGenerator.kt:663]: Yup

**Comment by yairm210** [comment]
Apart from minor nitpicks, looks good!
Remember to add the new map type to template.properties so it's translatable

**Comment by FiretronP75** [comment]
Feel free to nitpick all you want. I want to keep the code clean!

**Comment by FiretronP75** [comment]
OK I think it is ready.

**Review by yairm210 — APPROVED** [review]

**Pull request merged**